openquake/mbt/tools/mfd_sample/make_mfds.py

```diff
@@ -0,0 +1,277 @@
+        with open(fileout.format(cnum), 'wb') as f:
+            pickle.dump(catalogue, f)
+
+def _create_catalogue_versions(catfi, outdir, numcats=None, stype='random',
+                               numstd=1, rseed=122):
+    """
+    catfi: catalogue filename (pkl or hmtk/csv)
+    outdir: where to put the catalogues
+    numcats: number of catalogues to create, if stype is random
+    stype: 
+        random - samples from truncnorm
+        even - same sample for every magnitude; -1 : 1 by 0.2
+    rseed: random seed to control the cat generation
+    """
+    # check if output folder is empty
+
+    if os.path.exists(outdir):
+        tmps = f'\nWarning: {outdir} already exists! '
+        tmps += '\n Overwriting files.'
+        print(tmps)
+   #     sys.exit(1)
+    else:
+        os.makedirs(outdir)
+
+    csvout = os.path.join(outdir, 'v{}_catalogue.pkl')
+    fileout = os.path.join(outdir, 'v_mags.csv')
+    factors = np.arange(-1,1,0.1)
+
+    # read catalogue
+    if 'pkl' in catfi:
+        format = 'pickle'
+        catalogue = pd.read_pickle(catfi) 
+
+    elif ('csv' in catfi) or ('hmtk' in catfi):
+        parser = CsvCatalogueParser(catfi) # From .csv to hmtk
+        catalogue = parser.read_file()
+        format = 'hmtk'
+
+    else:
+        print(sys.stderr, "Use a supported catalogue format.")
+        sys.exit(1)
+
+    data = catalogue.data
+
+    if numcats and stype == 'even': 
+        print(sys.stderr, \
+              "Cannot specify number of catalogues for even sampling.")
+        sys.exit(1)
+
+    if stype == 'random': 
+
+        mags = []
+
+        time_strt = time.time()
+        for ii in np.arange(0,len(data['magnitude'])): 
+            m=data['magnitude'][ii]
+            sd=data['sigmaMagnitude'][ii]
+            allm = _get_truncated_normal(mean=m, sd=sd, 
+                                         low=m-numstd*sd, upp=m+numstd*sd)
+            mags_perm = allm.rvs(size=numcats, random_state=rseed)
+            mags.append(mags_perm)
+
+        marr = np.array(mags)
+        np.savetxt(fileout, marr, delimiter=',', fmt='%.2f')
+
+        for ii, ms in enumerate(marr.T): 
+            time_strt = time.time()
+            catalogue = deepcopy(catalogue)
+            catalogue.data['magnitude'] = np.array(ms)
+            _write_cat_instance(catalogue, format, csvout, ii)
+            time_end = time.time()
+
+
+    elif stype == 'even':
+        full_mags = {}
+        for jj, f in enumerate(factors):
+            mags = [f * ms + m for m, ms in zip(data['magnitude'], data['sigmaMagnitude'])]
+            catalogue = deepcopy(catalogue)
+            catalogue.data['magnitude'] = np.array(mags)
+            _write_cat_instance(catalogue, format, fileout, jj)
+        full_mags[jj] = mags
+        pd.DataFrame(full_mags).to_csv(fileout, index=None)
+
+    else:
+        print(sys.stderr, "Use a supported sampling type.")
+        sys.exit(1)
+
+
+
+def _decl_all_cats(outdir, cat, dcl_toml_tmp, decdir):
```

Comment:
is cat actually used here? It looks like we load in from a pkl in the outdir and then loop through, where cat becomes the iterator for catvs replacing any initial cat input